mantidproject/mantid
# ISIS Powder Pearl Seperated spline coefficient for long_mode

**Description of work.**
Made two separate values for the Spline coefficient and monitor spline coefficient for in and out of  too setup for later adding hard-coded defaults to long_mode.

**To test:**
Run the following script (note this assumes access to the ISIS data search archive)
```
from isis_powder import Pearl

config_file_path = r"/path/to/pearl_shared_config.yaml"
Pearl_example = Pearl(config_file=config_file_path,user_name="test",long_mode=False)
Pearl_example.create_vanadium(run_in_cycle=101508)
Pearl_example.create_vanadium(run_in_cycle=101508, long_mode=True)
```
with the following [files](https://github.com/mantidproject/mantid/files/2400213/PEARL.test.zip) (make sure that the spline_coefficient is set in the pearl_shared_config)
when long_modes value changes  you should see a message stating that the spline_coefficient has been changed back to its default value, 60.

 (note this is because currently long_mode on and off share the same default value, this will change in future)

Fixes #23504 
---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
